web/src/hooks/use-market-items.hook.js

```diff
@@ -1,36 +1,47 @@
-import {atomFamily, selectorFamily, useRecoilState} from "recoil"
-import {fetchMarketItems} from "../flow/fetch-market-items.script"
-import {IDLE, PROCESSING} from "../global/constants"
+import {atomFamily, useRecoilState} from "recoil"
+import useSWR from "swr"
+import {IDLE, LOADING} from "../global/constants"
+
+const fetcher = url => fetch(url).then(res => res.json())
 
 export const $state = atomFamily({
   key: "market-items::state",
-  default: selectorFamily({
-    key: "market-items::default",
-    get: address => async () => fetchMarketItems(address),
-  }),
+  default: [],
 })
 
 export const $status = atomFamily({
   key: "market-items::status",
   default: IDLE,
 })
 
-export function useMarketItems(address) {
-  const [items, setItems] = useRecoilState($state(address))
-  const [status, setStatus] = useRecoilState($status(address))
+export function useMarketItems() {
+  const url = process.env.REACT_APP_API_MARKET_ITEMS_LIST
+  const [status, setStatus] = useRecoilState($status(IDLE))
+  const [items, setItems] = useRecoilState($state([]))
+
+  useSWR(url, fetcher, {
+    initialData: items,
+    refreshInterval: 10,
+    errorRetryCount: 10,
+    onLoadingSlow: () => {
+      setStatus(LOADING)
+    },
+    onSuccess: ({latestSaleOffers}) => {
+      setStatus(IDLE)
+      setItems(latestSaleOffers)
+    },
+    onError: error => {
+      console.log("Failed to fetch market items.", error)
+    },
+  })
```

Comment:
Maybe move this into a separate file for API functions?

I see that this line is defined above, and would probably need to be copied through the project if we start making more API calls:

```js
const fetcher = url => fetch(url).then(res => res.json())
```